### PR TITLE
Add support for 2-level provisioning profile API

### PR DESCRIPTION
**Pull request by KrauseFx**
More information https://github.com/fastlane/fastlane/issues/6137

### CODE REVIEW

File: spaceship/lib/spaceship/portal/provisioning_profile.rb [hunk 1]
```diff
@@ -391,6 +390,19 @@ def managed_by_xcode?
       def mac?
         platform == 'mac'
       end
+
+      # Since 15th September 2016 this hidden behind another request
+      # see https://github.com/fastlane/fastlane/issues/6137 for more information
+      def certificates
+        @intenal_certificates ||= (details["certificates"] || []).collect do |cert|
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/portal/provisioning_profile.rb]: Nope

**Comment by maschall** [comment]
👍 This is working for me, with and without --skip-certificate-verification

**Review by maschall — APPROVED** [review]

File: spaceship/spec/provisioning_profile_spec.rb [hunk 2]
```diff
@@ -197,7 +198,7 @@
     let(:profile) { Spaceship::ProvisioningProfile.all.first }
 
     it "updates an existing profile" do
-      expect(client).to receive(:repair_provisioning_profile!).with('2MAY7NPHRU', 'net.sunapps.7 AppStore', 'store', '572XTN75U2', ["C8DL7464RQ"], [], mac: false).and_return({})
+      expect(client).to receive(:repair_provisioning_profile!).with('2MAY7NPHRU', 'net.sunapps.7 AppStore', 'store', '572XTN75U2', ["3BH4JJSWM4"], [], mac: false).and_return({})
```
Comment (maschall) [inline spaceship/spec/provisioning_profile_spec.rb]: we should use cert_id here too.

File: spaceship/lib/spaceship/portal/provisioning_profile.rb (hunk 1 above)
Comment (mfurtak) [inline spaceship/lib/spaceship/portal/provisioning_profile.rb]: typo `internal_certificates`

File: spaceship/spec/provisioning_profile_spec.rb [hunk 3]
```diff
@@ -159,35 +159,36 @@
...
       profile.certificates = []
-      expect(client).to receive(:repair_provisioning_profile!).with('2MAY7NPHRU', 'net.sunapps.7 AppStore', 'store', '572XTN75U2', ["C8DL7464RQ"], [], mac: false).and_return({})
+      expect(client).to receive(:repair_provisioning_profile!).with('2MAY7NPHRU', 'net.sunapps.7 AppStore', 'store', '572XTN75U2', [cert_id], [], mac: false).and_return({})
       profile.repair!
     end
 
     it "update the certificate if the current one is invalid" do
-      expect(profile.certificates.first.id).to eq('XC5PH8D47H') # this was the previous one
-      expect(client).to receive(:repair_provisioning_profile!).with('2MAY7NPHRU', 'net.sunapps.7 AppStore', 'store', '572XTN75U2', ["C8DL7464RQ"], [], mac: false).and_return({})
+      expect(profile.certificates.first.id).to eq('3BH4JJSWM4') # this was the previous one
```
Comment (mfurtak) [inline spaceship/spec/provisioning_profile_spec.rb]: replace with `cert_id`?

**Review by mfurtak — APPROVED** [review]
Approved with small possible improvements noted 👍

**Pull request merged**